python/tests/res/enkf/test_ert_templates.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright (C) 2017  Equinor ASA, Norway.
+#
+#  The file 'test_res_config.py' is part of ERT - Ensemble based Reservoir Tool.
+#
+#  ERT is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+from ecl.util.test import TestArea
+from res.enkf import ErtTemplates
+from res.enkf import ConfigKeys
+from res.enkf import ResConfig
+import unittest
+from tests import ResTest
+import os
+import copy
+
+
+class ErtTemplatesTest(ResTest):
+
+    def setUp(self):
+        self.work_area = TestArea("ert_templates_test_tmp")
+        self.config_data = {
+            ConfigKeys.CONFIG_DIRECTORY: self.work_area.get_cwd(),
+            ConfigKeys.CONFIG_FILE_KEY: "config",
+            ConfigKeys.RUN_TEMPLATE: [
+                ("namea", "filea", [("keyaa", "valaa"), ("keyab", "valab"), ("keyac", "valac")]),
+                ("nameb", "fileb", [("keyba", "valba"), ("keybb", "valbb"), ("keybc", "valbc")]),
+                ("namec", "filec", [("keyca", "valca"), ("keycb", "valcb"), ("keycc", "valcc")])
+            ]
+        }
+        self.filename = self.config_data[ConfigKeys.CONFIG_FILE_KEY]
+
+        # write a config file in order to load ResConfig
+        self.make_config_file(self.filename)
+        self.make_empty_file("namea")
+        self.make_empty_file("nameb")
+        self.make_empty_file("namec")
+        self.res_config = ResConfig(user_config_file=self.filename)
+
+    def tearDown(self):
+        del self.work_area
+
+    def test_all_templates_exist_with_correct_properties(self):
+        templates = ErtTemplates(self.res_config.subst_config.subst_list, config_dict=self.config_data)
+        template_names = templates.getTemplateNames()
+        configured_target_files = [t[0] for t in self.config_data[ConfigKeys.RUN_TEMPLATE]]
+        assert all(x in template_names for x in configured_target_files)
+        self.assertEqual(len(template_names), len(configured_target_files))
+        for configured_template in self.config_data[ConfigKeys.RUN_TEMPLATE]:
+            template = templates.get_template(configured_template[0])
+            self.assertEqual(template.get_template_file(), os.path.join(self.work_area.get_cwd(), configured_template[0]))
```

Comment:
maybe a __equal__ operator in templates could be created?

Reply:
This is checking that the configuration results in correct template objects, so comparison isn't going to work I think. Or did I misunderstand you?

Reply:
I think I just misread the code, as I thought configured_template was a template_type object